File: create-dmg/support/ApplescriptRunner.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int main(int argc, char *argv[]) {
    if (argc < 3) {
        fprintf(stderr, "Usage: %s <applescript_file> <handler_name> [args...]\n", argv[0]);
        return 1;
    }
    
    char *applescript_file = argv[1];
    char *handler_name = argv[2];
    
    // Build the osascript command
    char command[2048];
    snprintf(command, sizeof(command), "osascript \"%s\"", applescript_file);
    
    // Add handler name and any additional arguments
    for (int i = 3; i < argc; i++) {
        char arg_part[256];
        snprintf(arg_part, sizeof(arg_part), " \"%s\"", argv[i]);
        strncat(command, arg_part, sizeof(command) - strlen(command) - 1);
    }
    
    // Execute the command
    int result = system(command);
    return WEXITSTATUS(result);
}

File: create-dmg/support/applescript_runner.m
#import <Foundation/Foundation.h>
#import <OSAKit/OSAKit.h>

int main(int argc, char *argv[]) {
    @autoreleasepool {
        if (argc != 3) {
            fprintf(stderr, "Usage: %s <script_file> <function_name>\n", argv[0]);
            return 1;
        }
        
        NSString *scriptPath = [NSString stringWithUTF8String:argv[1]];
        NSString *functionName = [NSString stringWithUTF8String:argv[2]];
        
        // Read the AppleScript file
        NSError *error = nil;
        NSString *scriptSource = [NSString stringWithContentsOfFile:scriptPath 
                                                           encoding:NSUTF8StringEncoding 
                                                              error:&error];
        if (!scriptSource) {
            fprintf(stderr, "Error reading script file: %s\n", [[error localizedDescription] UTF8String]);
            return 1;
        }
        
        // Create and compile the AppleScript
        NSDictionary *errorInfo = nil;
        OSAScript *script = [[OSAScript alloc] initWithSource:scriptSource language:[OSALanguage languageForName:@"AppleScript"]];
        if (![script compileAndReturnError:&errorInfo]) {
            fprintf(stderr, "Error compiling script: %s\n", [[errorInfo description] UTF8String]);
            return 1;
        }
        
        // Execute the script
        NSAppleEventDescriptor *result = [script executeAndReturnError:&errorInfo];
        if (!result) {
            fprintf(stderr, "Error executing script: %s\n", [[errorInfo description] UTF8String]);
            return 1;
        }
        
        return 0;
    }
}
